boards/xtensa/intel_s1000_crb/intel_s1000_crb.dts

```diff
@@ -14,7 +14,16 @@
 		zephyr,sram = &sram0;
 		zephyr,console = &uart0;
 		zephyr,shell-uart = &uart0;
+		zephyr,flash = &flash0;
 	};
+
+#ifdef CONFIG_BOOTLOADER_MCUBOOT
```

Comment:
Is there a need for the ifdef?  Why not always define this?  I'm trying to remove CONFIG_ ifdef's from DTS so would prefer not to introduce a new one.

Reply:
Xtensa doesn't have XIP enabled right now. As a result, the binary should be copied from flash to SRAM for execution. That holds true for MCUBoot binary too. Since this MCUBoot binary in SRAM loads the actual Zephyr binary also into the same SRAM, it cannot overwrite itself while in execution (undefined behavior mostly leading to crash). That's why we have carved out a separate non-overlapping space for MCUBoot in the SRAM. However, there are scenarios (legacy) where S1000 can run without a bootloader. Hence the use of ifdef since we don't want to disrupt them.

Reply:
hmm.. is it possible to specify some overlay file in boards/xtensa/intel_s1000_crb/CMakeLists.txt and get it included in the build? Something like
if(CONFIG_BOOTLOADER_MCUBOOT)
       set(OVERLAY_FILE sram_address.dts)
endif()

Not sure how to include this in the build though (as it should be applicable only for this board).

Reply:
@galak.. I see that DTC_OVERLAY_FILE can be used to specify this. If i can somehow specify the sram_address.dts file to DTC_OVERLAY_FILE, this can be handled. Do you know how to update the DTC_OVERLAY_FILE variable from boards/xtensa/intel_s1000_crb/CMakeLists.txt?

Reply:
you just create an overlay file ${BOARD}.overlay in the app

Reply:
Yes. But the problem is it'll get added all the time. I want that file to be added only when CONFIG_BOOTLOADER_MCUBOOT is defined. Have already tried that.

Reply:
Also, i would prefer to add it in boards/xtensa/intel_s1000_crb since i don't want this info to be added to all the applications. Just an overhead and not scalable.

Reply:
I suggest a different approach for this, {Kconfig + linker script} instead of DTS overlay.
See `BOOTLOADER_SRAM_SIZE` in `$ZEPHYR_BASE/Kconfig.zephyr` and `include/arch/arm/cortex_m/scripts/linker.ld`

Currently `Kconfig.zephyr` has a `depends on ARM` for these and will need to be extended for xtensa.

Reply:
LGTM